com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java

```diff
@@ -108,17 +115,63 @@ protected Object resolveReferenceInContext(String key, Reference reference) {
     private Object resolveReference(Reference reference) {
         String referencedPropertyKey = reference.getPropertyName();
         Map<String, Object> replacementValues = valuesResolver.resolveProvidedValues(reference.getDependencyName());
-        if (replacementValues == null || !replacementValues.containsKey(referencedPropertyKey)) {
+
+        boolean canResolveInDepth = referencePattern.hasDepthOfReference() && referencedPropertyKey.contains("/");
+        if (replacementValues == null || (!replacementValues.containsKey(referencedPropertyKey) && !canResolveInDepth)) {            
             if (isStrict) {
                 throw new ContentException(Messages.UNABLE_TO_RESOLVE, getPrefixedName(prefix, referencedPropertyKey));
             }
             return null;
         }
-        String referencedPropertyKeyWithSuffix = getReferencedPropertyKeyWithSuffix(reference);
+        // always try to resolve as a flat reference first
         Object referencedProperty = replacementValues.get(referencedPropertyKey);
+        
+        if (referencedProperty == null && canResolveInDepth) {
+            referencedProperty = resolveInDepth(replacementValues, referencedPropertyKey);
+        }
+        
+        String referencedPropertyKeyWithSuffix = getReferencedPropertyKeyWithSuffix(reference);
         return resolve(referencedPropertyKeyWithSuffix, referencedProperty);
     }
-
+    
+    protected Object resolveInDepth(Map<String, Object> properties, String propertyKey) {
+        Matcher matcher = Pattern.compile("([^/]+)/?").matcher(propertyKey);
+        
+        if (!matcher.find()) {
+            if (isStrict) {
+                throw new ContentException(Messages.UNABLE_TO_RESOLVE, getPrefixedName(prefix, propertyKey));
+            }
+            return null;
+        }
+        
+        Object currentProperty = properties.get(matcher.group(1));
+        
+        while(matcher.find()) {
+            String subKey = matcher.group(1);
+            
+            if (StringUtils.isNumeric(subKey)) {

```

Comment:
It's possible for the sub key to be numeric and refer to a key in a map. So, first check what's the type of the referenced property and only validate whether the sub key is numeric in case it's a collection.